docs/porting/connectivity/Layer3IP.md

```diff
@@ -0,0 +1,131 @@
+
+To create and use an IP Layer3 interface, you need the `L3IPInterface` and `L3IP` classes. Moreover, you need an L3IP-derived driver class for your specific target device. Otherwise, the linker error `undefined reference to L3IP::get_default_instance()` stops build process.
+
+The user application code can create IP Layer3:
+
+   	NetworkInterface *l3interface;
+   	l3interface =new L3IPInterface(L3IP::get_default_instance(), OnboardNetworkStack::get_default_instance());
+   	l3interface->connect();
+
+This attaches the default network stack (usually LWIP - the other alternative is Nanostack) to the specified L3IP driver and provides all the `NetworkInterface` and `NetworkStack` APIs.
+
+Below is an example of a target device driver class that needs to be implemented:
+
+	class Cellular_driver_L3IP : public L3IP {... }
+
+	Cellular_driver_L3IP &Cellular_driver_L3IP::get_instance()
+	{
+    	static Cellular_driver_L3IP l3ip_test_driver;
+    	return l3ip_test_driver;
+	}
+
+	// Weak so a module can override
+	MBED_WEAK L3IP &L3IP::get_default_instance()
+	{
+    	return Cellular_driver_L3IP::get_instance();
+	}
+
+You can find this driver class example in the `mbed-os/TESTS/network/l3ip directory`.
+
+### Being the default interface 
+
+In Mbed OS, targets may provide default network interfaces through an automated factory method. You can do this with the following call:
+
+	NetworkInterface::get_default_instance()
+
+This way, you can create an ethernet, Wi-fi, mesh or cellular interface. The interface type depends on the `NETWORK_DEFAULT_INTERFACE_TYPE` value in the JSON configuration. Any interface you create this way is set as the default.
+
+IP Layer3 Interface doesn't use `NetworkInterface::get_default_instance`, so it is not set as the default when you create it.
+
+To set IP Layer3 as the default interface, you can use a new member of `set_as_default()`. This method is not limited to IP Layer3 and can set any network interface as default.
+
+### Cellular interfaces
+
+As a cellular interface, a little more work is necessary - at a minimum, you need to implement the extra configuration calls in `L3IPInterface`. This is because the network stacks and IP Layer3 APIs only relate to the basic data path - they have no knowledge of any other configuration mechanisms and assume they are already set up.
+
+To do this, create a C++ class that inherits from both `L3IPInterface` and a base class for the new IP Layer3 cellular interface solution. The `L3IPInterface` is a helper class that implements all the core `NetworkInterface` functionality for you. You then need to implement the extra configuration methods.
+
+You don't usually directly expose the `L3IP` driver class of a cellular driver - it is not usually declared as `L3IP::get_default_instance`. However, passing it to the constructor of your base `L3IPInterface` makes it visible using the `getl3ip` method. 
```

Comment:
Is this the first time we mention the `getl3ip` method? If so, could you please provide some context or a link?

Reply:
It's also  based on EMAC which has the same way of getting driver instance.
https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/porting/connectivity/ethernet.md
 
Both EMACInterface and L3IPInterface are helper classes provides glue logic between network drivers and network stack.
Methods getl3ip() is similar to  wifi/ethernet get_emac() and both are getting driver instance
getl3ip/get_emac  is a member of L3IPInterface /EMACInterface

Text replaced to:
" This then makes it visible using the `getl3ip` method which provides access to L3IP device driver instance. Method `getl3ip`is a member of `L3IPInterface` class."